kernel/syscall.c

```diff
@@ -1228,7 +1228,7 @@ static const char* _trim_trailing_slashes(
     {
         memcpy(buf, pathname, len + 1);
 
-        for (char* p = buf + len; p != buf && p[-1] == '/'; *--p = '\0')
+        for (char* p = buf + len; (p - buf) > 1 && p[-1] == '/'; *--p = '\0')
```

Comment:
Maybe it would be clearer to precheck at the start of the function whether the path is "/", like this:

```
    if (pathname[0] == '/' && pathname[1])
        return pathname;
```

Reply:
Also, this function already obtained the length of the pathname twice.

Reply:
Here is an alternative implementation.

```
static const char* _trim_trailing_slashes(
    const char* pathname,
    char* buf,
    size_t size)
{
    size_t len = strlen(pathname);

    if (len >= size)
        return NULL;

    /* if empty pathname or equal to "/" */
    if (len == 0 || (pathname[0] == '/' && pathname[1] == '\0'))
        return pathname;

    /* remove trailing slashes from the pathname if any */
    if (pathname[len - 1] == '/')
    {
        memcpy(buf, pathname, len + 1);

        for (char* p = buf + len; p != buf && p[-1] == '/'; *--p = '\0')
            ;

        pathname = buf;
    }

    return pathname;
}
```